### PR TITLE
Added big query source

### PR DESCRIPTION
**Pull request by jpjsonic**

### CODE REVIEW

File: .mint/test_visivo.yml [hunk 1]
```diff
@@ -115,6 +115,17 @@ tasks:
...
     filter: [test-projects/integration]
 
+  - key: test-bigquery
+    use: visivo-install
+    run: |
+      cd test-projects/integration
+      visivo run -s remote-bigquery
+    env:
+      STACKTRACE: true
+      BIGQUERY_BASE64_CREDENTIALS: ${{ secrets.BIGQUERY_BASE64_CREDENTIALS }}
```
Comment (tgsoverly) [inline .mint/test_visivo.yml:125]: Looks good

File: test-projects/integration/traces.visivo.yml [hunk 2]
```diff
@@ -99,7 +99,7 @@ traces:
       type: scatter3d
       x: query(x)
       y: query(y)
-      z: query( -2*(x%2)*x + x + y*(y-3))
+      z: query( -2*(x/16)*x + x + y*(y-3))
```
Comment (tgsoverly) [inline test-projects/integration/traces.visivo.yml:102]: Why did this change?

File: tests/models/sources/test_bigquery_source.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    data = {
+        "name": "source",
+        "database": "database",
+        "type": "bigquery",
+        "credentials_base64": "base64string", 
+        "project": "project-project"
+    }
+    source = BigQuerySource(**data)
+
+    assert "**********" in source.model_dump_json()
```
Comment (tgsoverly) [inline tests/models/sources/test_bigquery_source.py:25]: You probably copied what I had but maybe we could assert that it has the key and then the stars to make sure that the password field is the one that has the stars.

File: visivo/models/sources/bigquery_source.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        description="The Google Cloud service account credentials JSON string base64 encoded. Turn your JSON into a base64 string in the command line with `python -m base64 < credentials.json > encoded.txt`. ",
+    )
+
+    database: Optional[str] = Field(
+        None,
+        description="The default BigQuery dataset to use for queries.",
+    )
+
+    type: BigQueryType
+    connection_pool_size: Optional[int] = Field(
```
Comment (tgsoverly) [inline visivo/models/sources/bigquery_source.py:84]: I thought the connection pool size was already in the sequel alchemy

**Review by tgsoverly — APPROVED** [review]
Looks good a few non-blocking comments

File: test-projects/integration/traces.visivo.yml (hunk 2 above)
Comment (jpjsonic) [reply test-projects/integration/traces.visivo.yml:102]: Big query doesn't support the `%` as a modulus. The logic was arbitrary anyways so I switched it so we could easily add another integration test for bigquery

File: tests/models/sources/test_bigquery_source.py (hunk 3 above)
Comment (jpjsonic) [reply tests/models/sources/test_bigquery_source.py:25]: Yeah that's a good idea to get more specific.

File: visivo/models/sources/bigquery_source.py (hunk 4 above)
Comment (jpjsonic) [reply visivo/models/sources/bigquery_source.py:84]: I dont think so, unless I'm missing something

**Pull request merged**